### PR TITLE
Fwcfg: reboot the vm after install windbg

### PR DESCRIPTION
**Pull request by leidwang**
ID:2120153

Signed-off-by: Leidong Wang <leidwang@redhat.com>

### CODE REVIEW

**Comment by leidwang** [comment]
Hi @xiagao Could you help to review this patch?Thanks!

**Comment by leidwang** [comment]
Hi @xiagao Could you help to review this patch when your are free?Thanks!

**Comment by xiagao** [comment]
> Hi @xiagao Could you help to review this patch when your are free?Thanks!

Sure, I will do it before this Friday.

**Comment by xiagao** [comment]
Hi, just have a quick look at this patch. Could you add the commit log to make this patch more clear and easy understand?

**Comment by xiagao** [comment]
@leidwang  could you add a note before the code, give some info that it's just a workaround, and will remove it if there is a better way to avoid the error
e.g.     
            # xxxx
            if params.get("need_reboot", "no") == "yes":
            session = vm.reboot()

**Comment by xiagao** [comment]
The workaround looks good to me, just pls check my last comment.

**Comment by leidwang** [comment]
> The workaround looks good to me, just pls check my last comment.



> The workaround looks good to me, just pls check my last comment.

Okay, added.

File: qemu/tests/fwcfg.py [hunk 1]
```diff
@@ -67,6 +67,9 @@ def run(test, params, env):
         if not windbg_installed:
             win_dump_utils.install_windbg(test, params, session,
                                           timeout=wdbg_timeout)
+        # A temporary workaround for win2022
```
Comment (xiagao) [inline qemu/tests/fwcfg.py:70]: Could you make it more clear, such as you described in commit.

**Review by xiagao — APPROVED** [review]

**Comment by leidwang** [comment]
Hi @vivianQizhu Could you please help to review this patch?Thanks!

File: qemu/tests/fwcfg.py [hunk 2]
```diff
@@ -67,6 +67,9 @@ def run(test, params, env):
         if not windbg_installed:
             win_dump_utils.install_windbg(test, params, session,
                                           timeout=wdbg_timeout)
+        # A temporary workaround to clear up unexpected pop-up in guest
```
Comment (vivianQizhu) [inline qemu/tests/fwcfg.py:70]: # TODO: xxx

**Comment by xiagao** [comment]
@leidwang @vivianQizhu  The job passed this time without this workaround, I'm not sure if  it's totally resolved or probability problem, @leidwang did you run this case recently? However, we'd better not merge this patch right now.

**Comment by leidwang** [comment]
> @leidwang @vivianQizhu The job passed this time without this workaround, I'm not sure if it's totally resolved or probability problem, @leidwang did you run this case recently? However, we'd better not merge this patch right now.

Hi @xiagao 
The failure reproduce rate is not 100%, so the case can be passed sometimes,but when the pop-up appears, the failure will raised.So I add this workaround to the code.

**Comment by xiagao** [comment]
> > @leidwang @vivianQizhu The job passed this time without this workaround, I'm not sure if it's totally resolved or probability problem, @leidwang did you run this case recently? However, we'd better not merge this patch right now.
> 
> Hi @xiagao The failure reproduce rate is not 100%, so the case can be passed sometimes,but when the pop-up appears, the failure will raised.So I add this workaround to the code.

Thanks for your explanation, then it makes sense.

File: qemu/tests/fwcfg.py (hunk 2 above)
Comment (leidwang) [reply qemu/tests/fwcfg.py:70]: Hi @vivianQizhu 
This is just a note to document these codes are a workaround for unexpected popups, I don't think we need to add the TODO keyword. Not sure if I understand your suggestion clearly.

Comment (vivianQizhu) [reply qemu/tests/fwcfg.py:70]: @leidwang If it is a temporary workaround, I assume we need a 'TODO' to indicates that we want to fix it later.

Comment (leidwang) [reply qemu/tests/fwcfg.py]: Okay,updated the annotation.

Comment (leidwang) [reply qemu/tests/fwcfg.py]: Hi @vivianQizhu Could you please review it again?Thanks!

**Review by vivianQizhu — APPROVED** [review]

**Pull request merged**